### PR TITLE
Made Primitive static factories vararg

### PR DESCRIPTION
**Pull request by AlparSzabados**
Fixes issue: https://github.com/javaslang/javaslang/issues/1544

### CODE REVIEW

File: javaslang/src/test/java/javaslang/collection/ArrayTest.java [hunk 1]
```diff
@@ -57,42 +57,42 @@ protected boolean isDistinctElements() {
     }
 
     @Override
-    protected Array<Boolean> ofAll(boolean[] array) {
+    protected Array<Boolean> ofAll(boolean... array) {
```
Comment (paplorinc) [inline javaslang/src/test/java/javaslang/collection/ArrayTest.java]: I think these should be renamed too (and most other tests also)

File: javaslang/src/main/java/javaslang/collection/Array.java [hunk 2]
```diff
@@ -134,91 +134,99 @@ private Array(Object[] delegate) {
     }
 
     /**
-     * Creates a Array based on the elements of a boolean array.
+     * Creates an Array from boolean values.
```
Comment (paplorinc) [inline javaslang/src/main/java/javaslang/collection/Array.java]: feel free to correct the rest of `a Array`s too :)

File: javaslang/src/main/java/javaslang/collection/Array.java [hunk 3]
```diff
@@ -134,91 +134,99 @@ private Array(Object[] delegate) {
...
      *
-     * @param array a boolean array
+     * @param elements boolean values
      * @return A new Array of Boolean values
+     * @throws NullPointerException if elements is null
```
Comment (paplorinc) [inline javaslang/src/main/java/javaslang/collection/Array.java]: :+1:

**Comment by danieldietrich** [comment]
Hi Alpar, thank you - the PR looks great!
/cc Lorinc, thanks for the review!

File: javaslang/src/main/java/javaslang/collection/Array.java (hunk 2 above)
Comment (AlparSzabados) [inline javaslang/src/main/java/javaslang/collection/Array.java]: Done in: https://github.com/javaslang/javaslang/pull/1766

File: javaslang/src/test/java/javaslang/collection/ArrayTest.java (hunk 1 above)
Comment (AlparSzabados) [inline javaslang/src/test/java/javaslang/collection/ArrayTest.java]: Done in: https://github.com/javaslang/javaslang/pull/1766

**Pull request merged**